src/Components/Facility/AddLocationForm.tsx

```diff
@@ -45,9 +52,8 @@ export const AddLocationForm = (props: LocationFormProps) => {
     async function fetchFacilityName() {
       setIsLoading(true);
       if (facilityId) {
-        const res = await dispatchAction(getAnyFacility(facilityId));
-
-        setFacilityName(res?.data?.name || "");
+        const facility = await dispatchAction(getAnyFacility(facilityId));
+        setFacilityName(facility?.data?.name || "");
```

Comment:
use useQuery

Reply:
Khavin, the dispatch is currently inside the function which is inside the useEffect hook. There are multiple instances of this in the repo, so I will open a new PR fixing all the occurences

Reply:
The plan was to, all new codebase would not use `useDispatch` and would use `useQuery`. We are doing file by file rather than occurrence by occurrence as it'd be easier to test than if refactored by occurrence.